### ISSUE TITLE
Refactor `ComponentSelector` to match sklearn's `DecisionTreeClassifier`

### ISSUE THREAD

**Issue by tsalo** [ME-ICA/tedana#1034]
### Summary

If we use [`DecisionTreeClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html) as the general template for `ComponentSelector`.

- `__init__()` takes hyperparameters, but not data. In our case, that would be the `tree`, but not the `comptable`.
- `fit()` takes training data and fits the model. In our case, `comptable` is training data.
- `predict()` takes test data and predicts labels. In our case, `comptable` is the test data as well.

If we organize the tree's attributes similarly to `DecisionTreeClassifier`, we might be able to directly use sklearn's [`plot_tree`](https://scikit-learn.org/stable/modules/generated/sklearn.tree.plot_tree.html).
It would also be really cool if we could figure out how to plot the decision surfaces based on features (see [this example](https://scikit-learn.org/stable/auto_examples/tree/plot_iris_dtc.html#sphx-glr-auto-examples-tree-plot-iris-dtc-py)).
